subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -473,7 +474,8 @@ static int open(const struct device *instance)
 	rpmsg_inst->bound_cb = bound_cb;
 	rpmsg_inst->cb = ept_cb;
 
-	err = ipc_rpmsg_init(rpmsg_inst, data->role, data->vr.shm_io, &data->vr.vdev,
+	err = ipc_rpmsg_init(rpmsg_inst, data->role, conf->buffer_size,
```

Comment:
... and the more I look at this, the less I understand it.

It looks like `data->vr.shm_io` is a `struct metal_io_region*`, which agrees with the signature for the third parameter to `ipc_rpmsg_init()`, but `buffer_size` is just an int. In the above example in nrf5340dk_nrf5340_cpuapp.overlay, it's getting set to `128`, so it doesn't look like its an int that's been casted from a pointer.

What's going on?